core/lib/spree/core/controller_helpers/store.rb

```diff
@@ -26,7 +26,11 @@ def ensure_current_store(object)
           return if object.nil?
 
           if object.has_attribute?(:store_id)
-            object.store = current_store
+            if object.store.present?
```

Comment:
@MatthewKennedy let's add this as a validation method included in https://github.com/spree/spree/blob/master/core/app/models/concerns/spree/single_store_resource.rb

Reply:
@MatthewKennedy also let's add to the if condition `object.store != current_store`